### PR TITLE
Fixing tabbing logic in tab header component. 

### LINKED ISSUE #19580: A11y_Oracle Assessment for Azure Data Studio (ADS)_AssessmentReport_SkuRecommendations_Keyboard:Focus order for tab items is incorrect at "SKU Recommendations"
“[Check out Accessibility Insights! ](https://accessibilityinsights.io/)- Identify accessibility bugs before check-in and make bug fixing faster and easier.”

### GitHubTags:
 #A11yMAS;#A11yTCS;#A11ySev2;#BM_SQLAzureDataStudio_Win32_Jun2021;#SQL Azure Data Studio;#Win11;#DesktopApp;#WCAG2.4.3;#AINotInScope;#FTP;#Keyboard;

### Environment Details:
Application Name: Oracle Assessment for Azure Data Studio (ADS)
Operating System: Windows 11
Version :1.35.1

### Steps to Reproduce:

1. Download the azure data studio.
2. Open the application and tab to the Connections icon in left navigation.
3. Tab to the new connections button and press enter. Connection dialog box will open.
4. Fill Correct details in connection details and tab to the connect button and press enter. New connection will be added.
5. Tab to the Database Migration Assessment for Oracle button and then press enter. "Database Migration Assessment for Oracle" page will be opened.
6. Tab to the "Run New Assessment" button at top the page and press enter.New Database Migration Assessment for Oracle dialog box will open. Fill the correct details. Tab to "Start Assessment button and press enter.
7. New Assessment can be seen in last five assessment tab. Tab to the assessment press enter. New tab will get open.
8. Tab to Assessment overview by platforms section. Tab to Assessment for Azure virtual machine in the section.Tab to view report details button and press enter. New screen will open.
9. Tab to "Sku Recommendations" tab and press enter.
10. Navigate using keyboard and observe the tab items.

### Actual:
While navigating with keyboard after selecting the "SKU Recommendations" tab item , Keyboard focus is not shifting to the data of selected tab item instead its moving to the "Feature compatability"  tab item. 

### Expected:
While navigating with keyboard after selecting the "SKU Recommendations" tab item , Next Tab focus should be shifting to the data of selected tab item.

### User Impact:
Keyboard users will face difficulty to navigate in the selected tab item.

### PR DESCRIPTION
**Pull request by aasimkhan30**
This PR fixes #19580

Initial State:
image

After Pressing tab:
image

Pressing right arrow from initial state:
image

Code from: 
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/tab_role#example

### CODE REVIEW

File: src/sql/base/browser/ui/panel/panel.component.ts [hunk 1]
```diff
@@ -345,14 +346,17 @@ export class PanelComponent extends Disposable implements IThemable {
 
 	private focusNextTab(): void {
 		const currentIndex = this.focusedTabHeaderIndex;
+		this._tabHeaders.toArray()[currentIndex].nativeElement.tabIndex = -1;
```
Comment (Charles-Gagnon) [inline src/sql/base/browser/ui/panel/panel.component.ts:349]: Aren't you already doing this in the template?

`[attr.tabindex]="tab.active ? '0': '-1'"`

Comment (aasimkhan30) [reply src/sql/base/browser/ui/panel/panel.component.ts:349]: This is called when we use arrow keys to change focus. We do not change the active tab in that case.

Comment (Charles-Gagnon) [reply src/sql/base/browser/ui/panel/panel.component.ts:349]: Yeah, I agree with @alanrenmsft. We should be updating the active state when the active tab changes regardless of whether it's through the arrow keys, clicking or whatever. Is that not happening or something?

**Comment by aasimkhan30** [comment]
@Charles-Gagnon, can you please review this PR?

**Comment by aasimkhan30** [comment]
I have changed the code to only set the tabindex when creating a new tabHeader.  After that, all the tab index changes are controlled through code.

**Comment by Charles-Gagnon** [comment]
Right now this has different behavior when clicking vs using arrow navigation : 

When you click on a tab it doesn't change the tabindex with the result that the "old" tab is still the one with tabindex=0

TabClick

But when you use the arrow keys to navigate, then it'll update the tabindex to whatever one you last moved to, so that's the one that will be focused when tabbing in

TabArrow

The idea of what you're doing is fine, but it seems to me like there's some other fixes that need to be done around making sure that the active state of the tab header is updated correctly. Right now it seems like active is only set once upon creation and then never updated afterwards, so I'd suggest two things : 

1. Move the tabindex logic into the tab header component template so it automatically updates whenever the active state is changed
2. Update active as the focus changes

This should simplify things a bit and make it behave more consistently. 

Note that right now "active" seems to be referring to the selected tab - which isn't really correct. The "active" element on the page is just the one that has focus, where as selected would be better used to indicate what one is actually the one displaying its content. So if you're going to use "active" to indicate focus then you'd need to update that as well.

**Comment by aasimkhan30** [comment]
The first image is a faulty behavior. However, the second image is definitely the correct behavior. You can see an example over here: 

https://ms.portal.azure.com/#create/Microsoft.SQLDatabase

According to me there are 2 different states 
1. Active: The opened tab with the bottom blue border.
2. Focused: A tab where the current keyboard/mouse focus is. A focused tab has a blue outline. 


A tab can be in 

1. Active/Focused state: Both bottom blue border and outline is visible. The tab index of tab header  is 0  and rest all are -1
2. Active/Not Focused state: It only has the outline. The tab index of tabheader is 0 and rest all are -1
3. Inactive/Not Focused state: Tab index is -1. It has neither the bottom blue border nor the outline. 


What's wrong with my current code: 
A mouse click should made the clicked tab header's tab index to 0 and the rest all headers should have tab index to -1.

**Comment by Charles-Gagnon** [comment]
Yes, that sounds correct. In case it wasn't clear though I'm suggesting that we update the terms used here though - active is currently being used to indicate what tab is the one being shown on the screen but in HTML land usually refers to the currently focused element : https://developer.mozilla.org/en-US/docs/Web/API/Document/activeElement

Changing it so that we use active/selected instead will make this much clearer - where active is focused and selected is content shown.

**Comment by aasimkhan30** [comment]
Closing this PR as I have opened a new one #19770

**Pull request closed without merging**